Add Wang weights for offset detector scans

bnel1201 opened the pull request:
Hello Ander,

I saw this originally from issue [# 151](https://github.com/CERN/TIGRE/issues/151)  and thought I would try implementing [Ge Wang's paper](https://aapm.onlinelibrary.wiley.com/doi/epdf/10.1118/1.1489043). The following works with my projection data as well as with simulations from the demo data in d14_Offsets.m.  You mentioned in your comments to #151 that this issue is specific to FDK so I think this added control flow should be sufficient. This is my first time contributing to this project and I'm to work with you if any thing needs to be changed to better fit the project  workflow.

Other notes:
Similar to the original paper, the following works for up to 30% offset, beyond that I get some shading in the center. It's likely pixel re-binning may be necessary for offsets between 30% and the max 50%.

AnderBiguri commented:
@bnel1201 ah this is fantastically unexpected! Thanks a lot for doing this! Let me give it a try myself tomorrow and I will merge it if there is no issue (there should not be!)

I assume this resolves #151 and #101 then!

AnderBiguri commented on MATLAB/Algorithms/FDK.m:
```diff
@@ -21,6 +21,11 @@
 % Coded by:           Kyungsang Kim, modified by Ander Biguri 
```
Add your name here (if you want)

AnderBiguri commented on MATLAB/Algorithms/FDK.m:
```diff
@@ -116,4 +121,23 @@
+    overlap_in_pix = round(overlap_in_mm / geo.dDetector(1));
+
+    t_in_mm = linspace(-overlap_in_mm, overlap_in_mm, 2 * overlap_in_pix);
+    R_in_mm = geo.DSO;
```
While geometries where DSO changes per-projection should not be used with FDK, they are still supported and they may give us a bug in the future if someone tries to run this with that. I suggest changing this to 

```
R_in_mm = geo.DSO(1);
```

Not necessarily correct for changing DSOs, but neither is using FDK and Wang weights, so whatever. It solves our possible bug. maybe we can leave a comment mentioning that.

bnel1201 replied:
I completely agree, it worked for my situation but I'll fix this so it doesn't mess up any's code down the line.

AnderBiguri replied:
Great! Feel free to fix all these, otherwise I might find some time next week to apply them myself, if you can't do it. ;)

AnderBiguri commented on MATLAB/Algorithms/FDK.m:
```diff
@@ -116,4 +121,23 @@
+
+    wang_weights = ones(size(proj));
+    wang_weights(:, 1:size(w, 2), :) = w;
+    if sign(geo.offDetector(1)) < 0
```
Similar to the comment above (and not only here), we need to think about what happens if someone adds a varying offDetector (per projection). Do we compute `overlap_in_mm` for each offset, or do we just use the first value? 

I genuinely don't know, so I am happy to get input in here. If we just take the first value, then maybe its worth adding a warning in the cases that there are more informing the user that only the first will be used.

AnderBiguri commented:
> Similar to the original paper, the following works for up to 30% offset, beyond that I get some shading in the center. It's likely pixel re-binning may be necessary for offsets between 30% and the max 50%.

maybe we can also add a warning (or open a github issue) for this, so users know that this can happen? What do you think?

bnel1201 commented:
> > Similar to the original paper, the following works for up to 30% offset, beyond that I get some shading in the center. It's likely pixel re-binning may be necessary for offsets between 30% and the max 50%.
> 
> maybe we can also add a warning (or open a github issue) for this, so users know that this can happen? What do you think?

For example this is what I get with 34% displacement.
![34percent](https://user-images.githubusercontent.com/21038704/88235324-407b0a00-cc40-11ea-932c-db31353a2e4d.png)

I'll add a warning in the code if the offset exceeds 30%. But if a user gets this warning it would be nice to at least point them in the right direction so adding an issue would be a good idea as well. I have a rebinning function that I can clean up and contribute later. I know  [tomopy](https://tomopy.readthedocs.io/en/latest/api/tomopy.misc.morph.html#:~:text=misc.morph.sino_360_to_180%20%28data%2C%20overlap%20%3D%200%2C%20rotation%20%3D%20%27left%27%29,is%20odd%2C%20the%20last%20projection%20will%20be%20discarded.) has this re-binning available as sino_360_to_180 as an example we could point to.

bnel1201 commented:
Thanks for all the help! I think the most recent commit addresses your concerns. I'm not sure how to handle the pre-weighting when detector offsets or DSOs are changing per-projection so I added checks that only when these are constant the pre-weighting is applied. If those checks aren't satisfied then the pre-weighting is skipped and the usual FDK behavior continues.

AnderBiguri commented:
@bnel1201 I have a couple of tiny changes extra that I would like to add (couple of warnings, nothing to functionality). Is it OK if I push a commit to `bnel1201:master`?

bnel1201 commented:
Please do!
